### PR TITLE
278 unique query idenitifer

### PR DESCRIPTION
**Pull request by bbakerman**
As per #278  - This provides a unique id in the execution of a graphql query.  A later PR will provide the ability for api consumers to provide their own ExecutionIds but this gets you 90% on the way

### CODE REVIEW

File: src/main/java/graphql/Assert.java [hunk 1]
```diff
@@ -5,14 +5,14 @@
 public class Assert {
 
 
-    public static void assertNotNull(Object object, String errorMessage) {
```
Comment (bbakerman) [inline src/main/java/graphql/Assert.java]: So much more natural

   Obj o = notNull(input,"please")

File: src/main/java/graphql/GraphQL.java [hunk 2]
```diff
@@ -27,6 +28,18 @@
...
     private final ExecutionStrategy mutationStrategy;
+    //
+    // later PR changes will allow api consumers to provide their own id provider
+    //
+    // see https://github.com/graphql-java/graphql-java/pull/276 for the builder pattern
+    // needed to make this sustainable.  But for now we will use a hard coded approach.
+    //
+    private final ExecutionIdProvider idProvider = new ExecutionIdProvider() {
+        @Override
+        public ExecutionId generate(String query, String operationName, Object context) {
```
Comment (dminkovsky) [inline src/main/java/graphql/GraphQL.java]: How about calling this `provide()` since this may providers may provide an id not by generating it?

**Comment by bbakerman** [comment]
Ok fair enough

File: src/main/java/graphql/GraphQL.java (hunk 2 above)
Comment (dminkovsky) [inline src/main/java/graphql/GraphQL.java]: For example if http://stackoverflow.com/a/24910594/741970

Comment (bbakerman) [inline src/main/java/graphql/GraphQL.java]: Done

**Pull request merged**